pkg/cluster/loadbalancerprofile.go

```diff
@@ -321,27 +317,51 @@ func getDesiredOutboundIPs(managedOBIPCount int, ipAddresses map[string]mgmtnetw
 }
 
 func (m *manager) createPublicIPAddresses(ctx context.Context, ipAddresses map[string]mgmtnetwork.PublicIPAddress, numToCreate int) error {
-	ch := make(chan createIPResult)
-	defer close(ch)
-	var errResults []string
-	// create additional IPs if needed
-	for i := 0; i < numToCreate; i++ {
-		go m.createPublicIPAddress(ctx, ch)
-	}
+	var wg sync.WaitGroup
+
+	resultCh := make(chan mgmtnetwork.PublicIPAddress)
+	errCh := make(chan error)
+	doneCh := make(chan bool)
 
+	m.processIPAddresses(ctx, numToCreate, resultCh, errCh, &wg)
+
+	go func() {
+		wg.Wait()
+		close(resultCh)
+		close(errCh)
```

Comment:
These channels should be closed after finishing `consumeResults`, I guess.

According to [this](https://gobyexample.com/closing-channels#:~:text=Reading%20from%20a%20closed%20channel,channel%20is%20closed%20and%20empty.)
> Reading from a closed channel succeeds immediately, returning the zero value of the underlying type.

Also accoding to [this](https://go.dev/tour/concurrency/5)
> A select blocks until one of its cases can run, then it executes that case. It chooses one at random if multiple are ready.

That means, while `resultCh` and `errCh` are closed and `doneCh` is not closed, `for` loop in `consumeResults` never stops and reads the zero value from those channels.
And also even after `doneCh` is closed, it's not guaranteed that it stops. 

I wrote a small example. https://go.dev/play/p/jXqoRK-5V4b
If you run the program a few times, you probably see there are some `0`s printed as output.

I suggest those `close(resultCh)` and `close(errCh)` go into `defer` block, and in this goroutine we should just close `doneCh`, which makes sure that after `doneCh` is handled in `consumeResults`, it closes the other two channels.